src/esp/sensor/SensorFactory.cpp

```diff
@@ -28,7 +31,16 @@ sensor::SensorSuite SensorFactory::createSensors(
     //   //NonVisualSensor Setup
     // }
   }
-  return sensorSuite;
 }
+
+void SensorFactory::deleteSensor(const sensor::Sensor& sensor) {
+  delete (&sensor.node());
+}
+
+void SensorFactory::deleteSensor(scene::SceneNode& node,
+                                 const std::string& uuid) {
+  delete (&node.getSubtreeSensorSuite().get(uuid).node());
```

Comment:
Currently std::map will throw an out of range exception, so it's up to the user to make sure that sensors are not prematurely deleted.

Reply:
> so it's up to the user to make sure

No, this is not a good idea. As a developer, one of our duties is to keep the system robust so that you should handle the illegal cases.
Here, the correct way is that if such a sensor (with uuid) does not exist, the code will just return directly (I never say using ASSERT), no harm to the system.